dask_cloudprovider/providers/gcp/instances.py

```diff
@@ -0,0 +1,302 @@
+            .list(project=self.projectid, zone=self.zone, filter=f"name={self.name}")
+            .execute()
+        )
+        self.gcp_inst = d
+
+        if not d.get("items", None):
+            print("FAILURE")
+            print(self.gcp_inst)
+            raise Exception(f"Missing Instance {self.name}")
+
+        return d["items"][0]["status"]
+
+    async def close(self):
+        self.cluster.compute.instances().delete(
+            project=self.projectid, zone=self.zone, instance=self.name
+        ).execute()
+
+
+class GCPScheduler(GCPInstance, SchedulerMixin):
+    """Scheduler running in a GCP instance."""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.command = "dask-scheduler --host 0.0.0.0"
+        self.name = "aa-dask-scheduler"  # common_name
+
+    async def start(self):
+        await super().start()
+        await self.start_scheduler()
+
+    async def start_scheduler(self):
+        self.cluster._log("Creating scheduler instance")
+        self.internal_ip, self.external_ip = await self.create_vm()
+        self.address = f"tcp://{self.external_ip}:8786"
+        await self.wait_for_scheduler()
+
+        # need to reserve internal IP for workers
+        # gcp docker containers can't see resolve ip address
+        self.cluster.scheduler_internal_ip = self.internal_ip
+        self.cluster.scheduler_external_ip = self.external_ip
+
+
+class GCPWorker(GCPInstance, WorkerMixin):
+    """Worker running in an GCP instance."""
+
+    def __init__(self, scheduler, *args, worker_command=None, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.init_worker(scheduler, *args, worker_command=worker_command, **kwargs)
+
+    async def start(self):
+        await super().start()
+        await self.start_worker()
+
+    def init_worker(self, scheduler: str, *args, worker_command=None, **kwargs):
+        self.scheduler = scheduler
+        self.worker_command = worker_command
+
+        print("# use internal address for workers")
+        self.name = f"dask-worker-{str(uuid.uuid4())[:8]}"
+        self.command = (
+            f"{self.worker_command} {self.cluster.scheduler_internal_ip}:8786"
+        )
+        print(self.command)
+
+    async def start_worker(self):
+        self.cluster._log("Creating worker instance")
+        self.internal_ip, self.external_ip = await self.create_vm()
+        self.address = self.external_ip
+
+
+class GCPCluster(VMCluster):
+    """Cluster running on GCP Instances."""
+
+    def __init__(
+        self,
+        name="dask-gcp-example",
+        zone=None,
+        machine_type=None,
+        projectid=None,
+        source_image=None,
+        docker_image=None,
+        ngpus=None,
+        gpu_type=None,
+        worker_command="dask-cuda-worker",
+        **kwargs,
+    ):
+        super().__init__(**kwargs)
+        self.compute = googleapiclient.discovery.build("compute", "v1")
```

Comment:
I'm initially having credentials issues and am seeing the following exception. We should probably catch this and raise something friendlier.

```
    @pytest.mark.asyncio
    async def test_init():
>       cluster = GCPCluster(asynchronous=True)

dask_cloudprovider/providers/gcp/tests/test_gcp.py:44: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
dask_cloudprovider/providers/gcp/instances.py:284: in __init__
    self.compute = googleapiclient.discovery.build("compute", "v1")
../../../miniconda3/envs/daskdev/lib/python3.8/site-packages/googleapiclient/_helpers.py:134: in positional_wrapper
    return wrapped(*args, **kwargs)
../../../miniconda3/envs/daskdev/lib/python3.8/site-packages/googleapiclient/discovery.py:276: in build
    return build_from_document(
../../../miniconda3/envs/daskdev/lib/python3.8/site-packages/googleapiclient/_helpers.py:134: in positional_wrapper
    return wrapped(*args, **kwargs)
../../../miniconda3/envs/daskdev/lib/python3.8/site-packages/googleapiclient/discovery.py:516: in build_from_document
    credentials = _auth.default_credentials(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

scopes = None, quota_project_id = None

    def default_credentials(scopes=None, quota_project_id=None):
        """Returns Application Default Credentials."""
        if HAS_GOOGLE_AUTH:
>           credentials, _ = google.auth.default(scopes=scopes, quota_project_id=quota_project_id)
E           TypeError: default() got an unexpected keyword argument 'quota_project_id'

../../../miniconda3/envs/daskdev/lib/python3.8/site-packages/googleapiclient/_auth.py:54: TypeError
```